### PR TITLE
Enable GitOps functionality for the Serverless domain

### PR DESCRIPTION
**Pull request by pPrecel**
**Description**

Changes proposed in this pull request:

- Upgrade CRDs and other needed resources
- Bump images
- Fix other small bugs observed during integration

**Related issue(s)**

[Issue #8959](https://github.com/kyma-project/kyma/issues/8959)

### CODE REVIEW

File: resources/cluster-essentials/files/functions.serverless.crd.yaml [hunk 1]
```diff
@@ -3,7 +3,6 @@ kind: CustomResourceDefinition
 metadata:
   annotations:
     controller-gen.kubebuilder.io/version: v0.2.4
-    "helm.sh/resource-policy": keep
```
Comment (aerfio) [inline resources/cluster-essentials/files/functions.serverless.crd.yaml:6]: @magicmatatjahu is this annotation still needed?

Comment (magicmatatjahu) [reply resources/cluster-essentials/files/functions.serverless.crd.yaml:6]: I would keep it for the next release.

File: components/function-controller/internal/controllers/serverless/config.go [hunk 2]
```diff
@@ -27,6 +27,7 @@ type BuildConfig struct {
 	RuntimeConfigMapName string            `envconfig:"default=dockerfile-nodejs-12"`
 	ExecutorArgs         []string          `envconfig:"default=--insecure;--skip-tls-verify;--skip-unused-stages;--log-format=text;--cache=true"`
 	ExecutorImage        string            `envconfig:"default=gcr.io/kaniko-project/executor:v0.22.0"`
+	RepoFetcherImage     string            `envconfig:"eu.gcr.io/kyma-project/function-build-init:305bee60"`
```
Comment (tgorgol) [inline components/function-controller/internal/controllers/serverless/config.go:30]:
```suggestion
	RepoFetcherImage     string            `envconfig:"default=eu.gcr.io/kyma-project/function-build-init:305bee60"`
```

Comment (pPrecel) [reply components/function-controller/internal/controllers/serverless/config.go:30]: mb :D Thanks!

**Review by piotrmsc — APPROVED** [review]

**Pull request merged**